Failing rclpy-to-rclcpp WString pub/sub over Connext on Windows

## Bug report

**Required Info:**

- Operating System:
  - Windows 10
- Installation type:
  - From source
- Version or commit hash:
  - Current master at f65f852432090d1f2431d0cf496cbf4e2006922a
- DDS implementation:
  - RTI Connext
- Client library (if applicable):
  - rclcpp & rclpy

#### Steps to reproduce issue
Run `rclpy`-to-`rclcpp` pub/sub tests using Connext on Windows:

```
colcon test --event-handlers=console_direct+ --packages-select test_communication --ctest-args " -R" " ^test_publisher_subscriber__rclpy__rclcpp__rmw_connext_cpp"
```

#### Expected behavior
All tests pass.

#### Actual behavior

Subscription to a WString publisher fails with:

```
18: [test_subscriber-24] r
18: INFO:test_subscriber-24-stderr:[test_subscriber-24] r
18: [test_subscriber-24] eceived message does not match any expected message
18: INFO:test_subscriber-24-stderr:[test_subscriber-24] eceived message does not match any expected message
18: INFO:test_publisher-23-stdout:[test_publisher-23]
18: [ERROR] [test_subscriber-24]: process has died [pid 3156, exit code 3221226505, cmd 'C:/Users/Administrator/nuclearsandwich_ws/build/test_communication/Release/test_subscriber_cpp.exe WStrings /test_time_21_23_20'].
18: ERROR:test_subscriber-24:process has died [pid 3156, exit code 3221226505, cmd 'C:/Users/Administrator/nuclearsandwich_ws/build/test_communication/Release/test_subscriber_cpp.exe WStrings /test_time_21_23_20'].
```

#### Additional information

This issue does not manifest itself on `rclcpp`-to-`rclcpp` nor `rclpy`-to-`rclpy` nor `rclcpp`-to-`rclpy` tests.

The process terminates almost silently. Looking at the `rclcpp` subscriber [implementation](https://github.com/ros2/system_tests/blob/master/test_communication/test/test_subscriber.cpp#L51), this suggests that a crash occurs during `rclcpp::shutdown`.

--- a/test_communication/test/test_subscriber.cpp
+++ b/test_communication/test/test_subscriber.cpp
@@ -49,8 +49,6 @@ rclcpp::SubscriptionBase::SharedPtr subscribe(
         ++index;
       }
       if (!known_message) {
-        fprintf(stderr, "received message does not match any expected message\n");
-        rclcpp::shutdown();
         throw std::runtime_error("received message does not match any expected message");
       }
 
@@ -141,7 +139,14 @@ int main(int argc, char ** argv)
     rclcpp::shutdown();
     return 1;
   }
-  rclcpp::spin(node);
+
+  try {
+    rclcpp::spin(node);
+  } catch (const std::exception & e) {
+    std::cerr << e.what() << std::endl;
+    rclcpp::shutdown();
+    return 1;
+  }
 
   auto end = std::chrono::steady_clock::now();
   std::chrono::duration<float> diff = (end - start);

--- a/test_communication/test/test_subscriber.cpp
+++ b/test_communication/test/test_subscriber.cpp
@@ -143,7 +143,7 @@ int main(int argc, char ** argv)
   try {
     rclcpp::spin(node);
   } catch (const std::exception & e) {
-    std::cerr << e.what() << std::endl;
+    fprintf(stderr, "%s\n", e.what());
     rclcpp::shutdown();
     return 1;
   }

